Add what's new in v2 page + some additional v2 information (#710)
chocolatey/docs: input/en-us/troubleshooting.md

To disable these options, right-click on the PowerShell-Logo in the top-left of your terminal window, then select 'Properties'.

Source: https://serverfault.com/questions/204150/sometimes-powershell-stops-sending-output-until-i-press-enter-why

> ### SecurityProtocol was set to (some value), resetting to SystemDefault

Reviewer: Can you capitalize the headers?

Author: I avoided it because there's a lot of headers here which aren't capitalised, but... sure, I suppose.

Author: Mmm... running back through all these... This references an exact error message, as is done a few times previously in this file. I'm not sure we should be forcing capitalisation of things like this which are more or less an excerpt of an error message? It _shouldn't_ affect searches either way but it may make them less obviously recognisable to the eye?

Reviewer: Nope. I completely agree. This is an error message so should be verbatim (as much as possible).

This warning can show up with a few different forms, such as:

* `SecurityProtocol was set to Tls12, resetting to SystemDefault`
* `SecurityProtocol was set to Tls, Tls11, Tls12, resetting to SystemDefault`

This warning indicates that a package script, hook script, or an extension changed the TLS security protocol, usually during a package install, uninstall, or upgrade.

> Be wary of any packages that do this: depending on what it was set to, the package may be explicitly allowing older TLS versions than the system is configured to allow by default.
> This can be a security hazard, especially if the package is installing software over the Internet.

Reviewer: ```suggestion
Ensure you understand the implications of this change and contact the package maintainer to inform them of their package behaviour.
```

In most cases, packages and extensions should not be modifying the security protocol.

> Rather, you should manage the allowed TLS versions through your operating system; as of v2.0.0, Chocolatey CLI will defer to the operating system configuration for the allowed TLS versions.

Reviewer: ```suggestion
Rather, you should manage the allowed TLS versions through your Operating System. Chocolatey CLI v2.0.0 and later, will defer to the Operating System configuration for the allowed TLS versions.
```

See the [TLS Best Practices documentation](https://learn.microsoft.com/en-us/dotnet/framework/network-programming/tls) provided by Microsoft for more detailed information.

> ## `choco upgrade all` fails to upgrade a package, which cascades into failing many more seemingly unrelated packages

Reviewer: Can you capitalize the heading?

Author: Sure

Author: Sorted, ty

> Due to the way dependency resolution works, packages with certain dependency specifications can occasionally trigger undesirable behaviour when using `choco upgrade all` if the dependencies fail to install.
> Typically this will be encountered with metapackages, for example `obs-studio` and `obs-studio.install`.
> Since `choco upgrade all` attempts to upgrade every installed package, any packages which fail to install may be retried as part of a later dependency chain.
> For example, `obs-studio.install` may be upgraded, fail, and then later be tried again as part of the dependency chain for `obs-studio` itself.
> As a result, Chocolatey CLI's packages may be in an inconsistent state.

Reviewer: ```suggestion
Typically this will be encountered with meta-packages, such as `obs-studio` which takes a dependency on `obs-studio.install`. When running `choco upgrade all` it will attempt to upgrade every installed package. Any packages which fail to install may be retried as part of a later dependency chain.
For example, `obs-studio.install` may be upgraded, fail, and then later be tried again as part of the dependency chain for `obs-studio` itself.
As a result, Chocolatey CLI's packages may be in an inconsistent state.
```

> This is typically characterised by a dependency resolution error message when performing package operations, which looks like this:

Reviewer: ```suggestion
This is typically characterised by a dependency resolution error message when performing package operations:
```

```cmd
Unable to resolve dependencies. 'obs-studio.install 29.0.2' is not compatible with 'obs-studio 29.1.1 constraint: obs-studio.install (>= 29.1.1)'.
```

> Typically this is resolvable by directly upgrading the affected dependency.
> In this example, you would `choco upgrade obs-studio.install` to resolve the missing depedency and restore proper functionality.
> If an environmental issue is causing the upgrade to fail, you would need to resolve that before attempting to upgrade the dependency again.
> 

Reviewer: ```suggestion
This is normally resolved by upgrading the affected dependency.
In this example, you would run `choco upgrade obs-studio.install` to resolve the issue.
```

### Example

The following scenario illustrates what can happen during an `upgrade all` where a dependency fails to install.
In this example, `obs-studio` installs, but `obs-studio.install` fails.

> Due to the dependency version range specified by `obs-studio`, the version of `obs-studio.install`that remains installed due to the upgrade failure does not satisfy the dependency requirement.

Reviewer: ```suggestion
Due to the dependency version range specified by `obs-studio`, the version of `obs-studio.install` that remains installed, due to the upgrade failure, does not satisfy the dependency requirement.
```

Author: Amended! Removed a few commas, though.

Author: Maybe parentheses... hmm. I'll fiddle a bit.

Reviewer: It reads correctly to me with the commas 😄

Author: It can but it's ambiguous I think so i went with parentheses for now, it reads like there's comma splicing with that many commas 😁

> To resolve the issue, we can address the cause of the upgrade failure and then call `choco upgrade obs-studio.install` to get the correct dependency version installed once again.

Reviewer: ```suggestion
To resolve the issue, we run `choco upgrade obs-studio.install` to get the correct dependency version installed.
```

> For packages that have a specific version dependency, you may want to use `--version` to specify an exact version to install in order to resolve the dependency error.

Reviewer: ```suggestion
For packages that have a specific version dependency, add the `--version` option and specify the exact version to install..
```
